tests/callbacks/test_callbacks.py

```diff
@@ -214,18 +200,12 @@ def on_test_end(self, trainer, pl_module):
 
 def test_early_stopping_no_val_step(tmpdir):
     """Test that early stopping callback falls back to training metrics when no validation defined."""
-    class ModelWithoutValStep(LightTrainDataloader, TestModelBase):
-
-        def training_step(self, *args, **kwargs):
-            output = super().training_step(*args, **kwargs)
-            loss = output['loss']  # could be anything else
-            output.update({'my_train_metric': loss})
```

Comment:
For example here. This is not just a refactoring, this is changing the intention of the test. By adding a custom key we test that early stopping works on the given key and not just the default loss key. If you reduce all tests to the same template model, we lose the testing of the edge cases. Please consider keeping the train step here.